velox/exec/Task.cpp

```diff
@@ -996,17 +966,29 @@ void Task::driverClosedLocked() {
 }
 
 bool Task::checkIfFinishedLocked() {
-  if ((numFinishedDrivers_ == numTotalDrivers_) && isRunningLocked()) {
-    if (taskStats_.executionEndTimeMs == 0) {
-      // In case we haven't set executionEndTimeMs due to all splits depleted,
-      // we set it here.
-      // This can happen due to task error or task being cancelled.
-      taskStats_.executionEndTimeMs = getCurrentTimeMs();
+  if (isRunningLocked()) {
```

Comment:
nit:
how about
```
if (!isRunningLocked()) {
   return false;
}
...
return true;
```

Reply:
This is better. Thank you for the suggestion. Will change.